src/cache/cache_las.c

```diff
@@ -810,9 +808,28 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
 err:
     /* Resolve the transaction. */
     if (local_txn) {
-        if (ret == 0)
+        if (ret == 0) {
             ret = __wt_txn_commit(session, NULL);
-        else
+            if (ret == 0)
+                /* Traverse the keys again to flag the updates inserted to lookaside. */
+                for (i = 0, list = multi->supd; i < multi->supd_entries; ++i, ++list) {
+                    if (list->onpage_upd == NULL)
+                        continue;
+
+                    /* Skip aborted transactions. */
+                    for (upd = list->onpage_upd->next; upd != NULL && upd->txnid == WT_TXN_ABORTED;
+                         upd = upd->next)
+                        ;
+
+                    /*
+                     * Flag the update as now in the lookaside file. It is enough to only flag the
+                     * first update that is inserted to las and we know that the older updates are
+                     * all in las as well.
+                     */
```

Comment:
I think we should set the flag on all updates in the history store - it will make debugging simpler, and could mean that we don't need to keep a local variable in places to remember whether we saw an update in the history store.

Reply:
What about flagging after commit or unflagging after rollback? Which one do you prefer? @agorrod

Reply:
I prefer to do it inline and undo on rollback - I think it's cleaner and you flagged that it isn't a problem for correctness.
We do need to be careful during rollback to not clear the flag unless it was set in this round of eviction, but I think saving where we got to in the first pass should be enough to make sure we do that correctly.

Reply:
OK, since everyone thinks it is better to flag the update inline. I'll do as suggested.

Reply:
How do we know where we have got to before this pass? We have to remember where we have got to for each key?

Reply:
I think we need to unflag every update on the chain as we have no way to tell the update is inserted by the previous reconciliation or the ongoing one. It is safe to unflag all of them. In the worst case, the next try will insert some of them again into las.

Reply:
If we insert some of them again into las, didn't it defeat the purpose of having the flag ?

Reply:
That only happens when we need to the rollback txn in error cases, which is very rare. @bvpvamsikrishna

Reply:
When you walk the update list on line 668 couldn't you save a pointer to the first update you see that was flagged in a previous iteration and then only rollback the flag up to that update?

Reply:
You need to save updates for every key on the page. I don't think that worth the effort. We can just unflag them all and keep the code simple giving the error case is very rare.

Reply:
Are we saving enough information in the memento structure to make the decision during rollback?

Reply:
We are not saving that and memento has been deleted by Alex. C.

Reply:
OK - could you create a follow on ticket - I'd like to review the decision. It would be nice if the flag accurately reflected the state on disk (and in the least I'd like something searchable in JIRA that tells what the expected behavior is).

Reply:
Created WT-5401